### DISCUSSION TITLE
Incrementing a number based on prefix of existing files

### DISCUSSION THREAD

**Discussion by windyScripts** [SilentVoid13/Templater#861] [Help]
I was trying to generate links to files with a prefix for the file's numbering in terms of that file type and couldn't quite get it to work. For instance, if I was making cooking recipe notes with names such as 1cr, 2 cr, 3 cr etc and I wanted to get the last prefix existing+1 (4 in this example) how would I go about it? 

I'm a code noob and I tried a simple while loop but that didn't work. I was incrementing a number and running tp.file.exists("[["+number+"rest-of-file-name]]").

**Comment by AB1908** [comment]
You'll want to run `app.vault.getAbstractFileByPath("path/to/folder")`. This will give you back a `TFolder`. This object should have a `.children` attribute whose length you could use.

```js
const tfolder = app.vault.getAbstractFileByPath("path/to/folder");
const childCount = tfolder.children.filter(t=>t.extension == "md").length;
```
should be the general idea.
